src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -42,6 +60,11 @@ public static string GetSectionKey(string path)
             return lastDelimiterIndex == -1 ? path : path.Substring(lastDelimiterIndex + 1);
         }
 
+        /// <summary>
+        /// Extracts the path corresponding to the parent node for a given path.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>The original path minus the last individual key found in it. Null if the original path corresponds to a top level node.</returns>
```

Comment:
Given that you used "segment" instead of "key" for the method above, we can changes the first sentence to be: 

`The original path minus the last segment...`